% Template for PLoS
% Version 3.5 March 2018
%
% % % % % % % % % % % % % % % % % % % % % %
%
% -- IMPORTANT NOTE
%
% This template contains comments intended
% to minimize problems and delays during our production
% process. Please follow the template instructions
% whenever possible.
%
% % % % % % % % % % % % % % % % % % % % % % %
%
% Once your paper is accepted for publication,
% PLEASE REMOVE ALL TRACKED CHANGES in this file
% and leave only the final text of your manuscript.
% PLOS recommends the use of latexdiff to track changes during review, as this will help to maintain a clean tex file.
% Visit https://www.ctan.org/pkg/latexdiff?lang=en for info or contact us at latex@plos.org.
%
%
% There are no restrictions on package use within the LaTeX files except that
% no packages listed in the template may be deleted.
%
% Please do not include colors or graphics in the text.
%
% The manuscript LaTeX source should be contained within a single file (do not use \input, \externaldocument, or similar commands).
%
% % % % % % % % % % % % % % % % % % % % % % %
%
% -- FIGURES AND TABLES
%
% Please include tables/figure captions directly after the paragraph where they are first cited in the text.
%
% DO NOT INCLUDE GRAPHICS IN YOUR MANUSCRIPT
% - Figures should be uploaded separately from your manuscript file.
% - Figures generated using LaTeX should be extracted and removed from the PDF before submission.
% - Figures containing multiple panels/subfigures must be combined into one image file before submission.
% For figure citations, please use "Fig" instead of "Figure".
% See http://journals.plos.org/plosone/s/figures for PLOS figure guidelines.
%
% Tables should be cell-based and may not contain:
% - spacing/line breaks within cells to alter layout or alignment
% - do not nest tabular environments (no tabular environments within tabular environments)
% - no graphics or colored text (cell background color/shading OK)
% See http://journals.plos.org/plosone/s/tables for table guidelines.
%
% For tables that exceed the width of the text column, use the adjustwidth environment as illustrated in the example table in text below.
%
% % % % % % % % % % % % % % % % % % % % % % % %
%
% -- EQUATIONS, MATH SYMBOLS, SUBSCRIPTS, AND SUPERSCRIPTS
%
% IMPORTANT
% Below are a few tips to help format your equations and other special characters according to our specifications. For more tips to help reduce the possibility of formatting errors during conversion, please see our LaTeX guidelines at http://journals.plos.org/plosone/s/latex
%
% For inline equations, please be sure to include all portions of an equation in the math environment.  For example, x$^2$ is incorrect; this should be formatted as $x^2$ (or $\mathrm{x}^2$ if the romanized font is desired).
%
% Do not include text that is not math in the math environment. For example, CO2 should be written as CO\textsubscript{2} instead of CO$_2$.
%
% Please add line breaks to long display equations when possible in order to fit size of the column.
%
% For inline equations, please do not include punctuation (commas, etc) within the math environment unless this is part of the equation.
%
% When adding superscript or subscripts outside of brackets/braces, please group using {}.  For example, change "[U(D,E,\gamma)]^2" to "{[U(D,E,\gamma)]}^2".
%
% Do not use \cal for caligraphic font.  Instead, use \mathcal{}
%
% % % % % % % % % % % % % % % % % % % % % % % %
%
% Please contact latex@plos.org with any questions.
%
% % % % % % % % % % % % % % % % % % % % % % % %

\documentclass[10pt,letterpaper]{article}
\usepackage[top=0.85in,left=2.75in,footskip=0.75in]{geometry}

% amsmath and amssymb packages, useful for mathematical formulas and symbols
\usepackage{amsmath,amssymb}

% Use adjustwidth environment to exceed column width (see example table in text)
\usepackage{changepage}

% Use Unicode characters when possible
\usepackage[utf8x]{inputenc}

% textcomp package and marvosym package for additional characters
\usepackage{textcomp,marvosym}

% cite package, to clean up citations in the main text. Do not remove.
\usepackage{cite}

% Use nameref to cite supporting information files (see Supporting Information section for more info)
\usepackage{nameref,hyperref}

% line numbers
\usepackage[right]{lineno}

% ligatures disabled
\usepackage{microtype}
\DisableLigatures[f]{encoding = *, family = * }

% color can be used to apply background shading to table cells only
\usepackage[table]{xcolor}

% array package and thick rules for tables
\usepackage{array}

\usepackage{color,soul}
\usepackage[normalem]{ulem}


\usepackage{mhchem}
\usepackage{mathtools}


% create "+" rule type for thick vertical lines
\newcolumntype{+}{!{\vrule width 2pt}}

% create \thickcline for thick horizontal lines of variable length
\newlength\savedwidth
\newcommand\thickcline[1]{%
  \noalign{\global\savedwidth\arrayrulewidth\global\arrayrulewidth 2pt}%
  \cline{#1}%
  \noalign{\vskip\arrayrulewidth}%
  \noalign{\global\arrayrulewidth\savedwidth}%
}

% \thickhline command for thick horizontal lines that span the table
\newcommand\thickhline{\noalign{\global\savedwidth\arrayrulewidth\global\arrayrulewidth 2pt}%
\hline
\noalign{\global\arrayrulewidth\savedwidth}}


% Remove comment for double spacing
%\usepackage{setspace}
%\doublespacing

% Text layout
\raggedright
\setlength{\parindent}{0.5cm}
\textwidth 5.25in
\textheight 8.75in

% Bold the 'Figure #' in the caption and separate it from the title/caption with a period
% Captions will be left justified
\usepackage[aboveskip=1pt,labelfont=bf,labelsep=period,justification=raggedright,singlelinecheck=off]{caption}
\renewcommand{\figurename}{Fig}

% Use the PLoS provided BiBTeX style
\bibliographystyle{plos2015}


% Remove brackets from numbering in List of References
\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother



% Header and Footer with logo
\usepackage{lastpage,fancyhdr,graphicx}
\usepackage{epstopdf}
%\pagestyle{myheadings}
\pagestyle{fancy}
\fancyhf{}
%\setlength{\headheight}{27.023pt}
%\lhead{\includegraphics[width=2.0in]{PLOS-submission.eps}}
\rfoot{\thepage/\pageref{LastPage}}
\renewcommand{\headrulewidth}{0pt}
\renewcommand{\footrule}{\hrule height 2pt \vspace{2mm}}
\fancyheadoffset[L]{2.25in}
\fancyfootoffset[L]{2.25in}
\lfoot{\today}

\usepackage{makecell}
\usepackage{gensymb}
\usepackage{rotating}
\usepackage{pdflscape}
\usepackage{multirow}


%% Include all macros below
\newcommand{\lorem}{{\bf LOREM}}
\newcommand{\ipsum}{{\bf IPSUM}}
\newcommand{\angstrom}{\mbox{\normalfont\AA}}

\newenvironment{changemargin}[2]{%
\begin{list}{}{%
\setlength{\topsep}{0pt}%
\setlength{\leftmargin}{#1}%
\setlength{\rightmargin}{#2}%
\setlength{\listparindent}{\parindent}%
\setlength\textit{emindent}{\parindent}%
\setlength{\parsep}{\parskip}%
}%
\item[]}{\end{list}}


%% END MACROS SECTION


\begin{document}
\vspace*{0.2in}

% Title must be 250 characters or less.
\begin{flushleft}
{\Large
\textbf\newline{Bayesian inference and comparison of stochastic transcription elongation models} % Please use "sentence case" for title and headings (capitalize only the first word in a title (or heading), the first word in a subtitle (or subheading), and any proper nouns).
}
\newline
% Insert author names, affiliations and corresponding author email (do not include titles, positions, or degrees).
\\
Jordan Douglas\textsuperscript{1,2},
Richard Kingston\textsuperscript{1},
Alexei J. Drummond\textsuperscript{1,2*}
\\
\bigskip
\textbf{1} School of Biological Sciences, University of Auckland, Auckland, New Zealand
\\
\textbf{2} Centre for Computational Evolution, School of Computer Science, University of Auckland, Auckland, New Zealand
\\
\bigskip


% Use the asterisk to denote corresponding authorship and provide email address in note below.
* alexei@cs.auckland.ac.nz


\end{flushleft}
% Please keep the abstract below 300 words
\section*{Abstract}

Transcription elongation can be modelled as a three step process, involving polymerase translocation, NTP binding, and nucleotide incorporation into the nascent mRNA. This cycle of events can be simulated at the single-molecule level as a continuous-time Markov process using parameters derived from single-molecule experiments. Previously developed models differ in the way they are parameterised, and in their incorporation of partial equilibrium approximations.

We have formulated a hierarchical network comprised of 12 sequence-dependent transcription elongation models. The simplest model has two parameters and assumes that both translocation and NTP binding can be modelled as equilibrium processes. The most complex model has six parameters makes no partial equilibrium assumptions. We systematically compared the ability of these models to explain published force-velocity data, using approximate Bayesian computation. This analysis was performed using data for the RNA polymerase complexes of \textit{ E. coli}, \textit{ S. cerevisiae} and Bacteriophage T7.


Our analysis indicates that the polymerases differ significantly in their translocation rates, with the rates in T7 pol being fast compared to \textit{ E. coli} RNAP and \textit{ S. cerevisiae} pol II. Different models are applicable in different cases. We also show that all three RNA polymerases have an energetic preference for the posttranslocated state over the pretranslocated state. A Bayesian inference and model selection framework, like the one presented in this publication, should be routinely applicable to the interrogation of single-molecule datasets.


% Please keep the Author Summary between 150 and 200 words
% Use first person. PLOS ONE authors please skip this step.
% Author Summary not valid for PLOS ONE submissions.   
\section*{Author summary}

Transcription is a critical biological process which occurs in all living organisms. It involves copying the organism's genetic material into messenger RNA (mRNA) which directs protein synthesis on the ribosome. Transcription is performed by RNA polymerases which have been extensively studied using both ensemble and single-molecule techniques (see reviews: \cite{herbert2008single, dangkulwanich2014molecular}). Single-molecule data provides unique insights into the molecular behaviour of RNA polymerases. Transcription at the single-molecule level can be computationally simulated as a continuous-time Markov process and the model outputs compared with experimental data. In this study we use Bayesian techniques to perform a systematic comparison of 12 stochastic models of transcriptional elongation. We demonstrate how equilibrium approximations can strengthen or weaken the model, and show how Bayesian techniques can identify necessary or unnecessary model parameters. We describe a framework to a) simulate, b) perform inference on, and c) compare models of transcription elongation.


\linenumbers

% Use "Eq" instead of "Equation" for equation citations.
\section*{Introduction}
Transcription is carried out by RNA polymerases: RNAP in \textit{Escherichia coli}, pol II in \textit{ Saccharomyces cerevisiae}, and T7 pol in Bacteriophage T7. It involves the copying of template double-stranded DNA (dsDNA) into single-stranded messenger RNA (mRNA). RNAP and pol II are  comprised of multiple subunits, and their catalytic subunits are homologous \cite{sweetser1987prokaryotic, sosunov2003unified}. In contrast, T7 pol exists as a monomer with a distinct sequence, and resembles  the \textit{ E. coli} DNA polymerase I \cite{sousa1993crystal}.  \par




Optical trapping experiments have been performed on the transcription elongation complex (TEC) from a variety of organisms \cite{abbondanzieri2005direct, adelman2002single, bai2007mechanochemical, hodges2009nucleosomal, galburt2007backtracking, skinner2004promoter, dulin2015elongation}. In a typical experimental setup, two polystyrene beads (around 600 nm in diameter) are tethered to the system; one attached to the RNA polymerase and the other to the \hl{DNA} \cite{abbondanzieri2005direct}. As transcription elongation progresses, the distance between the two beads increases and the velocity of a single TEC can be computed. Optical tweezers can be used to apply a force $F$ to the system (Fig ~\ref{fig1}).  \par

\begin{figure}[!h]
%\includegraphics[width=0.5\textwidth]{Images/Figure1.pdf}
\caption{\textbf{ Effect of an applied force on elongation velocity.}
(A) Optical trapping setup showing dsDNA being transcribed by RNA polymerase (grey ellipse) into mRNA. Two polystyrene beads are tethered to the system allowing the application of force using optical tweezers. An assisting load $F>0$ acts in the same direction as transcription (top) while a hindering load $F<0$ acts in the opposing direction (bottom). Figure not to scale. (B) \hl{Schematic depiction of the effect of applying a force on RNA polymerase. Due to the stochastic nature of transcription at the single-molecule level, each experiment yields a different distance-time trajectory, even under the same applied force.} }
\label{fig1}
\end{figure}




Single-molecule studies of the TEC have revealed that RNA polymerases progress in a discontinuous fashion \cite{abbondanzieri2005direct, neuman2003ubiquitous, davenport2000single, tolic2004diversity, abbondanzieri2005picocalorimetry} with step sizes that correspond to the dimensions of a single nucleotide ($3.4$ $\angstrom$ \cite{watson1953molecular}). Consequently, at the single molecule level, transcription is best modelled as a discrete process rather than a continuous one.    \par



A single cycle in the main transcription elongation pathway (Fig ~\ref{fig2}) requires (1) Forward translocation of the RNA polymerase, making the active site accessible; (2) Binding of the complementary nucleoside triphosphate (NTP); (3) Addition of the nucleotide onto the 3$^\prime$ end of the mRNA. This third step involves NTP hydrolysis. Nucleoside monophosphate is added onto the chain and pyrophosphate is released from the enzyme.  \\

\begin{figure}[!h]
%\includegraphics[width=\textwidth]{Images/Figure2.pdf}
\caption{\textbf{ State diagrams of RNA polymerase.}
(A) The model of the main transcription elongation pathway, which shows the postulated states; the pathways for interconversion; and the rate constants that govern each part of the reaction. The transcription bubble is the set of $\beta_1 + h + \beta_2$ bases \hl{(see main text for definitions)} in the double-stranded DNA which are unpaired. States are denoted by $S(l, t)$ where $l$ is the length of the mRNA and $t$ is the position of the polymerase active site (small grey rectangle) with respect to the $3^\prime$ end of the mRNA.  Polymerase translocation displaces the polymerase by a distance of $\delta = 1$ bp = $3.4\; \angstrom$. During polymerisation the chain is extended by one nucleotide. (B) Instantiated posttranslocated state of RNA polymerase transcribing the \textit{rpoB} gene sequence, with $\beta_1 = 2, h = 9, \beta_2 = 1$. \hl{Forward translocation requires melting two T/A basepairs (right arrows). Backward translocation requires melting two C/G basepairs (left arrows).} The mRNA secondary structure would also require reconfiguration \cite{tadigotla2006thermodynamic, maoileidigh2011unified}.}
\label{fig2}
\end{figure}




Our study aimed to identify the best model to describe this reaction cycle for RNAP, pol II, and T7 pol, based on analysis of published force-velocity data. As there are three reactions, up to six rate constants may be necessary for a kinetic model of a single nucleotide addition. These describe forward and backwards translocation ($k_{fwd}$ and $k_{bck}$), binding and release of NTP ($k_{bind}$ and $k_{rel}$), and NTP catalysis and reverse-catalysis ($k_{cat}$ and $k_{rev}$), also known as pyrophosphorolysis \cite{maitra1967role}. However fewer than six parameters may be required in \hl{practice}. \\


First, it is reasonable to assume that polymerisation is effectively irreversible  \cite{erie1992single, rhodes1974ribonucleic, maoileidigh2011unified, bai2004sequence}, as pyrophosphorolysis is a highly exergonic reaction, reducing the number of rate constants to five.
Second, translocation between the pretranslocated and posttranslocated states, and/or NTP binding, may occur on timescales significantly more rapid than the other steps, in which case they may be modelled as equilibrium processes. These assumptions simplify the model, as the respective forward and reverse reaction rate constants are subsumed by a single equilibrium constant. Third, thermodynamic models of nucleic acid structure can be used to estimate sequence-dependent translocation rates $k_{fwd}(l)$ and $k_{bck}(l)$, \hl{by invoking transition state theory}, and this can sometimes result in parameter reduction \cite{tadigotla2006thermodynamic, maoileidigh2011unified, bai2004sequence}. \\





Irrespective of equilibrium assumptions and parameterisation, transcription elongation under applied force can be modelled in two fundamentally distinct ways. First, there are the \textbf{deterministic}  equations which can be used to calculate the mean pause-free elongation velocity $v(F, \text{[NTP]})$ as a function of force $F$ and NTP concentration [NTP]. \hl{ This kind of model can be derived from  the differential equations describing the time evolution of all species, by application of the steady state approximation. Force effects on the translocation step are incorporated using transition state theory} \cite{bustamante2004mechanical,cleland1975partition}.

An example is the following 3-parameter model \cite{abbondanzieri2005direct}.


\begin{eqnarray}
\label{eq:deterministicBoltzmann}
v(F, \text{[NTP]}) = \frac{k_{cat}}{1 + \frac{K_D}{\text{[NTP]}}(1 + K_\tau  e^{-F \delta / k_B T})}
\end{eqnarray}



where $\delta$ is the distance between adjacent basepairs (3.4 $\angstrom$, \cite{watson1953molecular}), $K_D = \frac{k_{rel}}{k_{bind}}$ is the equilibrium constant of NTP binding, $K_\tau = \frac{k_{bck}}{k_{fwd}}$ is the equilibrium constant of translocation, $k_B$ is the Boltzmann constant, and $T$ is the absolute temperature. Increasingly complex equations may be used as more parameters or states are added to the model \cite{abbondanzieri2005direct, maoileidigh2011unified, bai2007mechanochemical}. Such equations describe the velocity averaged across an ensemble of molecules. Parameter inference applied to velocity-force-[NTP] experimental data is straightforward and computationally fast when using these equations. However these equations do not describe the distribution of velocity nor do they account for site heterogeneity across the nucleic acid sequence and therefore cannot predict local sequence effects. \par


Second, there are the \textbf{stochastic} models, which can be implemented via simulation of single-molecule behaviour using the Gillespie algorithm \cite{gillespie1977exact}. The mean velocity can be calculated by averaging velocities over a number of simulations for a given $F$ and [NTP]. This offers not just the mean but a full distribution of velocities and could potentially explain emergent properties unavailable from a deterministic model. Unfortunately, simulating can be very slow and therefore parameter inference can be a problem. \\


In this study we used a Markov-chain-Monte-Carlo approximate-Bayesian-computation (MCMC-ABC) algorithm \cite{beaumont2010approximate} to estimate transcription elongation parameters for \textbf{stochastic} models via simulation. The observed pause-free velocities we are fitting to were measured at varying applied force and NTP concentration. For each RNA polymerase under study -- \textit{ E. coli} RNAP, \textit{ S. cerevisiae} pol II, and T7 pol -- we fit to one respective dataset from the single-molecule literature \cite{abbondanzieri2005direct,schweikhard2014transcription,thomen2008t7}.




% Results and Discussion can be combined.
\section*{\hl{Models}}



\subsection*{Notation and state space}


Suppose the TEC is transcribing a gene of length $L$. Then let $S(l, t)$ denote a TEC state, where the mRNA is currently of length $l \leq L$, and $t \in \mathbb{Z}$ describes the position of the active site with respect to the $3^\prime$ end of the mRNA. When $t=0$ the polymerase is pretranslocated and cannot bind NTP, and when $t=1$ the polymerase is posttranslocated and \textit{can} bind NTP  (Fig ~\ref{fig2}). This study is focused on the main elongation pathway and the observed velocities being fitted have pauses filtered out. Therefore, although additional backtracked states ($t < 0$) \cite{wang1998force, shaevitz2003backtracking, abbondanzieri2005direct} and hypertranslocated states ($t > 1$) \cite{artsimovitch2000pausing, zhou2007dissociation} exist, these are not incorporated in the model.

\hl{Let $\beta_1$ and $\beta_2$ be the number of unpaired template nucleotides upstream and downstream of RNA polymerase, respectively, and let $h$ be the number of basepairs in the DNA/mRNA hybrid } (Fig ~\ref{fig2}A). \hl{ Although there are uncertainties in these parameters, they are held constant at $h = 9$, $\beta_1 = 2$, and $\beta_2 = 1$ } \cite{greive2005thinking, maoileidigh2011unified}.

\hl{ Transcription of the gene begins at state $S(l_0,0)$ and ends upon reaching $S(L, 0)$, where $l_0 = \beta_1 + h + 2$.}


\subsection*{Parameterisation of the NTP binding step}

\hl{// New section. Content relocated from Model Space}



NTP binding has been modelled as both a kinetic and equilibrium process in the literature \cite{bai2004sequence, abbondanzieri2005direct, maoileidigh2011unified}.

In a kinetic binding model, NTP binding occurs at pseudo-first order rate $k_{bind}$[NTP], while NTP release occurs at rate $k_{rel}$. In this case, $k_{bind}$ and $\frac{k_{rel}}{k_{bind}}$ must be estimated.

Under a partial equilibrium approximation NTP binding and release are assumed to be rapid enough that equilibrium is achieved. In this case, the rate constants $k_{bind}$ and $k_{rel}$ are subsumed by the NTP dissociation constant $K_D = \frac{k_{rel}}{k_{bind}}$ which becomes the sole binding-related parameter to estimate.



\subsection*{Parameterisation of the translocation step}





While inferences about the rate constants associated with NTP binding and catalysis ($k_{bind}$, $\frac{k_{rel}}{k_{bind}}$, and $k_{cat}$) can be made directly from the data, the translocation step is more complex. \hl{Transition state theory} is invoked in order to estimate $k_{fwd}$ and $k_{bck}$. Recasting the problem in this way (1) provides a way of accommodating the effects of applied force on the elongation process, and (2) allows the sequence-dependence of translocation to be incorporated by considering the energetics of basepairing. \hl{When allowing for sequence dependence, the total number of translocation rates required to model translocation of the full gene is $2(L - l_0)$.}




\subsubsection*{\hl{Thermodynamic models of base pairing energies}}

\hl{ // New section adapted from Nucleic Acid Thermodynamics (previously in Materials and Methods) and two paragraphs from the Introduction }

%The energetics of translocation are dependent on the position within the template.

The standard Gibbs free energies $\Delta_r G^0 ( = \Delta G)$  involved in duplex formation are calculated using nearest neighbour models. The standard Gibbs energy of state $S$ -- arising from nucleotide basepairing and dangling ends -- is calculated as


\begin{eqnarray}
    \Delta G^{(bp)}_{S} =\Delta G^{(bp)}_{gene} +  \Delta G^{(bp)}_{hybrid}
\end{eqnarray}

where SantaLucia's DNA/DNA basepairing parameters \cite{santalucia1998unified} are used to calculate $\Delta G^{(bp)}_{gene}$ and Sugimoto's DNA/RNA parameters \cite{wu2002temperature} are used for $\Delta G^{(bp)}_{hybrid}$. For the latter, dangling end energies are estimated as described by Bai et al. 2004 \cite{bai2004sequence}. Here, and elsewhere, the $^{(bp)}$ superscript is used to denote a model parameter that can be evaluated from the sequence alone. Gibbs energies are expressed relative to the thermal energy of the system,  in units of $k_BT$, where $k_BT = 0.6156$ kcal/mol at $T=310$ K. \par





In order for RNA polymerase to translocate forward (backward), up to two basepairs must be disrupted: (1) the basepair at the downstream (upstream) edge of the transcription bubble, and (2) the basepair at the upstream (downstream) end of the DNA/mRNA hybrid (Fig ~\ref{fig2}B). Differences in the basepairing energies in these regions confer sequence-dependence on the rate of translocation.

%For example in SantaLucia's parameters \cite{santalucia1998unified} 5$^\prime$ TA/AT is the weakest doublet at $-0.94$ $k_BT$ while 5$^\prime$ GC/CG is the strongest at $\Delta G^{(bp)} = -3.64$ $k_BT$.

\subsubsection*{\hl{Calculation of translocation rates or translocation equilibrium constant}}

\hl{ // New section. The critical information regarding translocation kinetics has been brought to the front so that minor details can be skipped by the reader}



The standard Gibbs energies of the pre and posttranslocated states, $\Delta G^{(bp)}_{S(l,0)}$  and  $\Delta G^{(bp)}_{S(l,1)}$, respectively, are used with up to four additional terms -- $\Delta G_{\tau 1}$, $\delta_1$, $\Delta G^\ddag_{\tau}$, and $\Delta G_{T(l,t)}^{(bp)}$ --  to calculate the translocation rates. The first three are model parameters which must be estimated while the latter is directly evaluated from the sequence.



Let $T(l,t)$ be the translocation transition state between $S(l,t)$ and $S(l,t+1)$. Then $\Delta G_{T(l,t)}^{\ddag} = \Delta G^\ddag_{\tau} + \Delta G_{T(l,t)}^{(bp)}$ is the sequence-dependent standard Gibbs energy of activation which must be overcome in order to translocate (Fig \ref{fig3}).


\begin{figure}[!h]
%\includegraphics[width=\textwidth]{Images/Figure3.pdf}
\caption{\textbf{Parameterisation of the translocation step.}
(A) Effects of model parameters on state energies. The figure displays a schematic Gibbs energy landscape of translocation, with backtracked states included for visualisation purposes. The solid red lines represent translocation states ($t=0$: pretranslocated, $t=1$: posttranslocated, and $t<0$: backtracked), while the dashed red lines represent transition states. Applying an assisting force $F>0$ tilts the landscape in favour of higher values of $t$. The effect of $\Delta G_{\tau 1}$ is observed at the posttranslocated state $t=1$. In a translocation equilibrium model, \hl{the barrier height is assumed to be so small, = translocation is so rapid, that the transition states are disregarded}.  \hl{(B) A model for the sequence-dependent transition state between translocation states $S(l,  0)$ and $S(l,  1)$. This is required for estimating the Gibbs energy of basepairing $\Delta G_{T(l,t)}^{(bp)}$ in the transition state. The basepairing energy, added to a baseline term $\Delta G^\ddag_{\tau}$, together specify the height of the activation barrier  } (Equation \ref{eq:translocationBarrier}). }
\label{fig3}
\end{figure}





Given an applied force $F$, the translocation rates governing transition between the pre and posttranslocated states ($k_{fwd}(l)$ and $k_{bck}(l)$) are calculated from barrier height $\Delta G_{T(l,0)}^{\ddag}$ using an Arrhenius type relation:


\begin{eqnarray}
%\label{eq:DGpostEquilibrium}
    k_{fwd}(l) = Ae^{-(\Delta G_{T(l,0)}^{\ddag} - \Delta G_{S(l,0)}^{(bp)} - F \delta_1 / k_BT )} \\
    k_{bck}(l) = Ae^{-(\Delta G_{T(l,0)}^{\ddag} - (\Delta G_{S(l,1)}^{(bp)} + \Delta G_{\tau 1}) + F (\delta - \delta_1) / k_BT)}
\end{eqnarray}

The derived rates $k_{fwd}(l)$ and $k_{bck}(l)$ are therefore dependent on the local sequence. The pre-exponential factor $A$ is held constant at $10^6$ s$^{-1}$. This term has been arbitrarily set to a variety of values in previous studies ($10^6 - 10^9$ s$^{-1}$ \cite{bai2004sequence, tadigotla2006thermodynamic, maoileidigh2011unified}). This has little consequence for model fitting, however the value of $\Delta G_{T(l,t)}^{\ddag}$ is entangled with the value of the pre-exponential factor $A$ and can only be meaningfully interpreted in light of its value.



If the system has time to reach equilibrium, the probabilities of observing the pretranslocated state $S(l,0)$ and posttranslocated state $S(l,1)$ are


\begin{eqnarray}
\label{eq:DGpost}
p(S(l,0)) \propto e^{-(\Delta G^{(bp)}_{S(l,0)})} \\
p(S(l,1)) \propto e^{-(\Delta G_{\tau 1} + \Delta G^{(bp)}_{S(l,1)})}
\end{eqnarray}



This is described by equilibrium constant $K_\tau$.

\begin{align}
\label{eq:DGpostEquilibrium}
K_\tau(l) &= \frac{p(S(l,0))}{p(S(l,1))} \\
&= exp\{ -(\Delta G_{S(l,0)} -  \Delta G_{S(l,1)} ) \} \\
 &= exp\{ -(\Delta G^{(bp)}_{S(l,0)} -  \Delta G^{(bp)}_{S(l,1)} - \Delta G _{\tau 1} ) \}
\end{align}



The physical meanings of the terms $\Delta G_{\tau 1}$, $\delta_1$, $\Delta G^\ddag_{\tau}$, and $\Delta G_{T(l,t)}^{(bp)}$, and the way they are used in the model, are detailed below.


\subsubsection*{Energetic bias for the posttranslocated states}
$\Delta G_{\tau 1}$ (units $k_BT$) is a parameter added to the standard Gibbs energy of the posttranslocated state. If $\Delta G_{\tau 1} = 0$, then the sequence alone determines the Gibbs energy difference between pre and posttranslocated states. In this case, pretranslocated states are usually favoured over posttranslocated states due to the loss of a single basepair in the hybrid of the latter.


$\Delta G_{\tau 1}$ has frequently been estimated for T7 pol \cite{yin2004structural, yu2012small, thomen2005unravelling} and there has been discussion around whether such a term is necessary for RNAP \cite{bai2007mechanochemical}. \\

\subsubsection*{Polymerase displacement and formation of the transition state}
$\delta_1$ (units $\angstrom$) is the distance that the polymerase must translocate forward to facilitate the formation of the transition state. The distance between adjacent basepairs is held constant at an experimentally measured value $\delta = 3.4\; \angstrom$ \cite{watson1953molecular}, and   $0 <\delta_1 < \delta$. The response of the system to an applied force $F$ depends on this term. In general, the application of force $F$ tilts the Gibbs energy landscape -- the Gibbs energy difference between adjacent translocation states being augmented by a factor $\frac{F\delta}{k_BT}$ (Fig~\ref{fig3}A, \cite{depken2009origin, herbert2008single}).



It may be necessary to estimate $\delta_1$ to model the data adequately \cite{maoileidigh2011unified}, or it may be sufficient to simply set $\delta_1 = \delta / 2$ \cite{depken2009origin}. \\


\subsubsection*{Energy barrier of translocation}


$\Delta G^\ddag_{\tau}$ and $\Delta G_{T(l,t)}^{(bp)}$ (units $k_BT$) together determine the activation barrier height in the translocation step. It is assumed that the sequence-dependent standard Gibbs energy of activation $\Delta G_{T(l,t)}^{\ddag}$ can be written as



\begin{eqnarray}
\label{eq:translocationBarrier}
    \Delta G_{T(l,t)}^{\ddag} = \Delta G^\ddag_{\tau} + \Delta G_{T(l,t)}^{(bp)}
\end{eqnarray}



$\Delta G^\ddag_{\tau}$ is therefore a sequence-independent baseline term used to compute the translocation barrier heights.  The parameter $\Delta G^\ddag_{\tau}$ must be estimated in order to evaluate translocation rates.

\hl{// The paragraph below was relocated and adapted from SI Appendix 3. SI Appendix 3 has since been deleted.}


In contrast $\Delta G_{T(l,t)}^{(bp)}$ is a term that is evaluated directly from the sequence derived from a model of the transition state (Fig \ref{fig3}B). The term is evaluated as the standard Gibbs energy of a TEC containing all hybrid and gene basepairs found in both $S(l,t)$ and $S(l,t+1)$, ie. the intersection between the two sets of basepairs.








\subsection*{Model space}

The full transcription elongation model makes use of the following 6 parameters:

\begin{itemize}
\item $k_{cat}$ (units s$^{-1}$).
\item $K_D = \frac{k_{rel}}{k_{bind}}$ (units $\mu$M).
\item $k_{bind}$ (units $\mu$M$^{-1}$ s$^{-1}$).
\item $\Delta G_{\tau 1}$ (units $k_BT$).
\item $\delta_1$ (units $\angstrom$).
\item $\Delta G^\ddag_{\tau}$ (units $k_BT$).
\end{itemize}


However fewer than 6 parameters may be needed to adequately describe the data. If it is assumed that the energy differences between pre and posttranslocated states are determined by basepairing energies alone, the parameter $\Delta G_{\tau 1}$ does not need to be estimated. This is equivalent to holding $\Delta G_{\tau 1}$ constant at 0. If it is assumed that the displacement required for formation of the translocation intermediate state is half the distance between adjacent basepairs, the parameter $\delta_1$ does not need to be estimated. This is equivalent to holding $\delta_1$ constant at $\delta/2$.

Partial equilibrium approximations may also simplify the model, as detailed above. If binding is approximated as an equilibrium process, $k_{bind}$ does not need to be estimated. If translocation is approximated as an equilibrium process,  $\Delta G^\ddag_{\tau}$ and $\delta_1$ do not need to be estimated. One, both, or neither of these two steps (binding and translocation) could be assumed to achieve equilibrium, thus yielding four equilibrium model variants (Fig ~\ref{fig4}A). The introduction of partial equilibrium approximations for both the NTP binding and translocation steps has implications when specifying the prior distributions for the Bayesian analysis (\nameref{S4_Appendix}.)
 \hl{The chemical master equations for single nucleotide addition cycles of these models are presented in} \nameref{S2_Appendix}. \\

\begin{figure}[!h]
%\includegraphics[width=\textwidth]{Images/Figure4.pdf}
\caption{\hl{// B: Arrows in figure now have labels } \textbf{ The space of models to be compared.} (A) The four equilibrium model variants. NTP binding, translocation, both, or neither, could be assumed to achieve equilibrium prior to catalysis. (B) The 12 transcription elongation models. An arrow connects model $i$ to $j$ if augmentation of model $i$ with a single parameter generates model $j$. The number of parameters to estimate  $k$ is shown for each level in the network. Equilibrium approximation colour scheme is the same as in A. $\Delta G_{\tau 1}$ and $\delta_1$ can each be estimated or set to a constant.}
\label{fig4}
\end{figure}



Incorporating these simplifications to the model in a combinatorial fashion results in a total of 12 related models, which together constitute the model space. Our objective was to determine which of these 12 models provides the best description of the experimental data. The simplest model (Model 1) contains 2 parameters ($k_{cat}$ and $K_D$). The most complex model (Model 12) contains all 6 parameters. The full model space is displayed in Fig ~\ref{fig4}B.






\subsection*{Stochastic modelling}


For each model we performed stochastic simulations, appropriate for the modelling of single-molecule force-velocity data. The simulations, performed using the Gillespie algorithm \cite{gillespie1977exact, lecca2013stochastic}, can be used to estimate the mean elongation velocity under a model.



The estimation of mean velocity can be broken down into three steps. First, the system is initialised by placing the RNA polymerase at the $3^\prime$ end of the template -- state $S(l_0, 0)$ -- with the transcription bubble open and a DNA/RNA hybrid formed. The force and NTP concentrations are assigned their experimentally set values. Second, a chemical reaction is randomly sampled. The probability that reaction $S \xrightarrow[]{k} S^\prime$ is selected is proportional to its rate constant $k$ (Fig \ref{fig2}).  The amount of time taken for the reaction to occur is sampled from the exponential distribution. States which are subject to a partial equilibrium approximation are coalesced into a single state, which augments the outbound rate constants. The second step is repeated until the RNA polymerase has copied the entire template. Third, the previous two steps are repeated $c$ times. The mean elongation velocity is evaluated as the mean of each mean elongation velocity across $c$ simulations. For further information, see \nameref{S1_Appendix}.\\






\subsection*{Relation to previous models and stochastic simulations}


\hl{// Section relocated and adapted from ``Discussion" }

There is an extensive literature concerned with the kinetic modelling of transcription elongation. Such models may incorporate backtracking, hypertranslocation, and other reactions. Here we are concerned only with the central elongation pathway.


A stochastic and sequence-dependent model  was proposed by Bai et al. 2004 \cite{bai2004sequence} for RNAP, with both NTP binding and translocation treated as equilibrium processes. The translocation equilibrium constant was calculated entirely from basepairing energies. Therefore this model is equivalent to Model 1, and the parameters were estimated as $k_{cat} = 24.7$ s$^{-1}$ and $K_D = 15.6$ $\mu M$ from fit to experimental data. Maoil{\'e}idigh et al. 2011 also presented stochastic simulations of RNAP. The elongation component of their model is equivalent to Model 6 \cite{maoileidigh2011unified}. We build on this work by providing a systematic Bayesian framework for model comparison and parameter estimation.   \par

While our analysis employed sequence-dependent stochastic models, comparisons can also be made with some deterministic models.


Abbondanzieri et al. 2005 \cite{abbondanzieri2005direct}, Larson et al. 2012 \cite{larson2012trigger} , Schweikhard et al. 2014 \cite{schweikhard2014transcription}, and Thomen et al. 2008. \cite{thomen2008t7, thomen2005unravelling}  described a deterministic model (for RNAP, pol II, pol II, and T7 pol respectively) which estimated $k_{cat}$, $K_D$ and translocation equilibrium constant $K_\tau = \frac{k_{bck}}{k_{fwd}}$. These are most similar to Model 4.




Maoil{\'e}idigh et al. 2011 for RNAP, and Dangkulwanich et al. 2013 for pol II, however found that the translocation and catalysis were occurring on similar timescales, and modelled only NTP binding as an equilibrium process  \cite{maoileidigh2011unified, Dangkulwanich2013complete}. They also estimated the distance of translocation. These deterministic models are most similar to Model 11.




Finally, Mejia et al. 2015 \cite{mejia2015trigger} used a model that is quite different to all the above models, as it does not explicitly treat translocation. Instead elongation is modelled with a two step kinetic scheme, the first step involving NTP binding and conformational change, and the second step involving nucleotide incorporation and product release. This model is most similar to a special case of Model 5 where $\Delta G_{\tau 1}$ becomes extremely negative, driving the polymerase into the posttranslocated position.  






\section*{\hl{Results and Discussion}}





\subsection*{Model selection with MCMC-ABC}


Our aim was to 1) use Bayesian inference to select the best of 12 transcription elongation models for each RNA polymerase; and 2) estimate the parameters for those of the models appearing in the 95\% credible set of the posterior distribution. Selecting prior distributions behind each parameter is a  critical process in Bayesian inference. A prior distribution should reflect what is known about the parameter before observing the new data. We have explicated our prior assumptions, with justifications, in Table \ref{table2}. \par


We performed MCMC-ABC experiments which estimated the parameters and model indicator $M_i$ for $i \in \mathbb{Z}, 1 \leq i \leq 12$. Models which appear more often in this posterior distribution are better choices, given the data. \par  



\hl{// The content below was relocated from the now-removed Materials and Methods}
The datasets we fit our models to are all from the single-molecule literature and are presented in: Figures 5a and 5b of Abbondanzieri et al. 2005 \cite{abbondanzieri2005direct} for \textit{ E. coli} RNAP, Figure 2a of Schweikhard et al. 2014 \cite{schweikhard2014transcription} for \textit{ S. cerevisiae} pol II, and Table 2 of Thomen et al. 2008 \cite{thomen2008t7} for T7 pol. To computationally replicate these experiments as faithfully as we could with the available information and computational limitations, simulations in this study were run on the 4 kb \textit{E. coli} \textit{rpoB} gene for RNAP (GenBank: EU274658), the first 4.75 kb of the human \textit{ rpb1} gene for pol II (NCBI: NG\_027747) the first 10 kb of the Enterobacteria phage $\lambda$ genome for T7 pol (NCBI: NC\_001416). The mean velocities from 32 (for RNAP), 10 (for pol II) and 3 (for T7 pol) simulations of the full respective sequences were used to estimate the mean elongation velocity during MCMC-ABC, given $F$ and [NTP].  \par

For further information about the MCMC-ABC algorithm \cite{beaumont2010approximate, csillery2010approximate}, see ~\nameref{S3_Appendix}.



\begin{table}[!htb]
\begin{adjustwidth}{-2.25in}{0in}
\centering
\caption{
\textbf{ Prior distributions used during Bayesian inference.}}
\resizebox{7.5in}{!}{ % Width = 5.25in + 2.25in


\begin{tabular}{|ccc|}
\hline
&&\\
\textbf{Parameter} & \textbf{Prior distribution(s)} & \textbf{Justification of prior distribution(s)} \\

&&\\
\hline
&&\\



Model $M$ &
\makecell{$P(M_i) = 2/16$ for $i \in \{1,2,4,5\}$ \\
 $P(M_i) = 1/16$ for $i \in \{3,6,7,8,9,10,11,12\}$} &
\makecell{ Each model should each have uniformly distributed values. \\ Models with translocation at equilibrium have double \\ the prior probability since these models do not use $\delta_1$.} \\

&&\\
\hline
&&\\



$k_{cat}$ (s$^{-1}$) & \makecell{ Lognormal($\mu = 3.454$, $\sigma = 0.587$) \textbf{for RNAP/pol II} \\ Lognormal($\mu = 4.585$, $\sigma = 0.457$) \textbf{for T7 pol} }  & \makecell{
 $k_{cat}$ and elongation velocity estimates for \textit{E. coli} RNAP \\ and \textit{ S. cerevisiae} pol II range from 18 to 50 s$^{-1}$  for optical \\ trapping  experiments \cite{bai2007mechanochemical, bai2004sequence, mejia2015trigger, galburt2007backtracking, hodges2009nucleosomal}, but as much as \\ 100 bp/s \textit{ in vivo} \cite{vogel1994rna, ryals1982temperature, richardson1996control, mason2005distinction}. Distribution selected such \\ that  $(10,100)$ is central 95\% interval. \\ For T7 pol $k_{cat}$ and elongation velocity estimates range \\ from 43 - 240 bp/s \cite{skinner2004promoter, iost1992bacteriophage, bonner1994characterization, anand2006transient}. Distribution selected such \\ that $(40,240)$ is central 95\% interval.}\\

&&\\
\hline
&&\\


$K_D$ ($\mu$M) & Lognormal($\mu = 1.844$, $\sigma = 1.762$)
 & \makecell{ Estimates for $K_D$ under binding equilibrium models range \\ from 20-140 $\mu$M \cite{rhodes1974ribonucleic, bai2007mechanochemical, kireeva2008transient, larson2012trigger, thomen2005unravelling}. In models where binding is \\ kinetic and slow, $K_D \equiv \frac{k_{rel}}{k_{bind}}$ could be much lower \\ (\nameref{S4_Appendix}). To accommodate for both binding models, \\ the prior distribution was selected such that the central \\ 95\% interval is (0.2, 200).} \\
&&\\
\hline
&&\\

$k_{bind}$ ($\mu$M$^{-1}$s$^{-1}$) & Lognormal($\mu = -1.498$, $S\sigma = 1.585$) & \makecell{
    Central 95\% interval set so that NTP binding is a slow \\ kinetic step (\nameref{S4_Appendix} ). Centered around (0.01, 5).  }  \\

&&\\
\hline
&&\\


$\Delta G_{\tau 1}$ ($k_B T$) & \makecell{ Normal($\mu = 0$, $\sigma = 1.55$) \textbf{for RNAP/pol II} \\ Normal($\mu = -3.3$, $\sigma = 1.55$) \textbf{for T7 pol}}  & \makecell{For RNAP and pol II, centered around 0 with a standard \\ deviation comparable to the free energy of a single nucleotide \\ basepair doublet, and such that the 95\% central interval is (-4, 4). \\ For T7 pol $\Delta G_{\tau 1}$ has been estimated as -4.3 \cite{thomen2005unravelling} and \\ -4.87   $k_BT$ \cite{yin2004structural}. However these estimates are likely resulting \\ partially from dangling ends. Thus, we subtracted the mean \\ dangling end contribution of $\sim$ -1 $k_BT$ \cite{santalucia1998unified} and centered the prior \\ around this interval with a standard deviation the same as above.
} \\

&&\\
\hline
&&\\

$\Delta G^\ddag_{\tau}$ ($k_B T$) & \makecell{ Normal($\mu = 5.5$, $\sigma = 0.97$) \textbf{for RNAP/pol II} \\ Normal($\mu = 2.5$, $\sigma = 1.36$) \textbf{for T7 pol} } & \makecell{ Central 95\% interval set so that translocation is a slow \\ kinetic step (\nameref{S4_Appendix} ). Selected so that 99\% central \\ interval is (3, 8) for RNAP and pol II, and (-1, 6) for T7 pol.}  \\

&&\\
\hline
&&\\



$\delta_1$ ($\angstrom$) & Uniform($l = 0$, $u = 3.4$) & Uniformly distributed across all possible values. \\

&&\\
\hline



\end{tabular}

}


\label{S2_Table}

\begin{flushleft} Prior distributions behind all estimated parameters and the model indicator. Unless specified otherwise, the prior distribution is used for all three RNA polymerases. Lognormal priors (parameterised in log space) are used for rates and equilibrium constants while normal priors are used for Gibbs energy terms. \hl{To maintain statistical integrity of the Bayesian analysis, prior distributions were not derived from the data presented by} Abbondanzieri et al. 2005 \cite{abbondanzieri2005direct} for RNAP, by Schweikhard et al. 2014 \cite{schweikhard2014transcription} for pol II, or by Thomen et al. 2008 \cite{thomen2008t7} for T7 pol.
\end{flushleft}
\label{table2}
\end{adjustwidth}
\end{table}









\subsection*{The posterior distributions}


\hl{// Section relocated and adapted from ``Model selection with MCMC-ABC" }

The posterior distributions from our MCMC-ABC experiments are presented in Table \ref{table1}, Fig ~\ref{fig5}, and Fig ~\ref{fig6}.


% Place tables after the first paragraph in which they are cited.
\begin{table}[!ht]
\begin{adjustwidth}{-2.25in}{0in} % Comment out/remove adjustwidth environment if table fits in text column.
\centering
\caption{
\textbf{ Summary of MCMC-ABC experiments.}}
\resizebox{7.5in}{!}{ % Width = 5.25in + 2.25in
%\begin{footnotesize}
\begin{tabular}{|cc|cc|cc|c|}
    \hline

 &&  \multicolumn{2}{c|}{} & \multicolumn{2}{c|}{} & \multicolumn{1}{c|}{} \\
&\textbf{Enzyme} &  \multicolumn{2}{c|}{\textit{ E. coli} RNAP} & \multicolumn{2}{c|}{\textit{ S. cerevisiae} pol II} & \multicolumn{1}{c|}{Bacteriophage T7 pol} \\
 &&  \multicolumn{2}{c|}{} & \multicolumn{2}{c|}{} & \multicolumn{1}{c|}{} \\
\hline

&&&&&& \\
&$\epsilon$ & \multicolumn{2}{c|}{2.39} & \multicolumn{2}{c|}{0.705} &  4.63 \\
&&&&&& \\
& Combined chain length & \multicolumn{2}{c|}{$3.5 \times 10^7$} & \multicolumn{2}{c|}{$6.2 \times 10^7$} &  $1.2 \times 10^8$ \\
&&&&&& \\

\hline

&&&&&& \\
&$i$ & 11 & 12 & 11 & 12 &  5 \\
%$M_i$ description & Binding eq & 12 & 8 & 11 & 12 & 4 & 5 \\
&&&&&& \\

Model & Description & \makecell{ Binding equilibrium, \\ Translocation kinetic } & \makecell{ Binding kinetic, \\ Translocation kinetic} & \makecell{ Binding equilibrium, \\ Translocation kinetic} & \makecell{ Binding kinetic, \\ Translocation kinetic} &   \makecell{ Binding kinetic, \\ Translocation equilibrium} \\

&&&&&& \\

\hline
&&&&&& \\




\multirow{12}{1.4cm}{ESS  / $\hat{R}$} &$\hat{k}_{cat}$ & 257 / 1.04   & 1441 / 1.03 & 549 / 1.02 & 1203 / 1.01  & 2110 / 1.00 \\
&&&&&& \\

 &$\hat{\frac{k_{rel}}{k_{bind}}}$  & 328 / 1.01   & 101 / 1.01 & 536 / 1.01 & 133 / 1.05  & 106 / 1.00 \\
&&&&&& \\


&$\hat{k}_{bind}$    & $-$   & 705 / 1.09 & $-$ & 516 / 1.02  & 154 / 1.00 \\
&&&&&& \\

&$\Delta \hat{G}_{\tau 1}$ & 466 / 1.02   & 1844 / 1.00 & 1145 / 1.00 & 2769 / 1.00  & 1626 / 1.02 \\
&&&&&& \\



&$\hat{\delta}_1$   & 300 / 1.04   & 2290 / 1.03 & 658 / 1.01 & 1469 / 1.00  & $-$ \\
&&&&&& \\

&$\Delta \hat{G}^\ddag_{\tau}$   & 340 / 1.02   & 1680 / 1.03 & 589 / 1.02 & 1179 / 1.00  & $-$ \\






&&&&&& \\
\hline

&&&&&& \\
Posterior &$P(M_i|D)$  & 0.81  & 0.19 & 0.29 & 0.71 & 0.96  \\
&&&&&& \\
\hline

\end{tabular}
%\end{footnotesize}
}
\begin{flushleft} \hl{// Rhat values and each individual ESS is now shown in this plot. The parameter estimates were moved to Fig 5}  Each column summarises the posterior distribution for the respective RNA polymerase, which arises from multiple independent MCMC chains. Approximate Bayesian computation threshold $\epsilon$ is shown for each enzyme; state $\Theta$ is accepted into the posterior distribution only if $X^2(\Theta) \leq \epsilon$ (\nameref{S3_Appendix}). Models which appear in an RNA polymerase's 95\% credible set and their posterior probabilities $P(M_i|D)$ are shown. The effective sample size (ESS, calculated with Tracer 1.6 \cite{rambaut2013tracer}) and R-hat ($\hat{R}$ \cite{gelman1992inference, brooks1998general, brooks2011handbook}) of each parameter, conditional on $M_i$, are displayed. A large ESS ($> 100$) and a small $\hat{R}$ ($< 1.1$) imply that the MCMC experiment has converged. Where a parameter is not incorporated in the kinetic model, a `$-$' is left in its place.
\end{flushleft}
\label{table1}
\end{adjustwidth}
\end{table}





\begin{figure}[!ht]
%\includegraphics[width=0.9\textwidth]{Images/Figure5.pdf}
\caption{\hl{\textbf{Posterior and prior distribution plots.}} Posterior distributions for all models which appear in the 95\% credible set are displayed (two models for RNAP, two models for pol II, and one model for T7 pol). Plots show the prior probability density $P(\theta)$ of each parameter and posterior probability density of each parameter conditional on the model $P(\theta | D, M_i)$. The geometric median point-estimates and  highest posterior density (HPD) intervals (calculated with Tracer 1.6 \cite{rambaut2013tracer}) are displayed above each plot (3 sf). }
\label{fig5}
\end{figure}





\begin{figure}[!h]
%\includegraphics[width=\textwidth]{Images/Figure6.pdf}
\caption{\hl{// Figure relocated from ``Supporting Information"} \textbf{Posterior distributions of simulated velocities.} Black open circles represent experimentally measured mean velocities reported in the original publication for (A) RNAP,  (B) pol II, and (C) T7 pol \cite{abbondanzieri2005direct,schweikhard2014transcription,thomen2008t7}. Each coloured dot represents a single sample simulated from the posterior distribution of parameters/models for the respective polymerase. 30 samples were generated from each of the three posterior distributions. For RNAP, [NTP]$_{eq}$ is defined as [ATP] = 5 $\mu$M, [CTP] = 2.5 $\mu$M, [GTP] = 10 $\mu$M, and [UTP] = 10 $\mu$M. }
\label{fig6}
\end{figure}









A large effective sample size ($>$ 100 \cite{rambaut2013tracer}) and a small $\hat{R}$ ($< 1.1$, as defined by Gelman et al. 1992  \cite{gelman1992inference, brooks2011handbook, brooks1998general}) are essential for making reliable parameter estimates. Table \ref{table1} suggests that the parameters in the 95 \% credible set of models are sufficiently estimated by these criteria.





These results indicate that the best models for the datasets examined are Models 11 and 12 for both RNAP and pol II, and Model 5 for T7 pol (Fig ~\ref{fig4}B).



For  pol II, Model 12 has the highest posterior probability $P(M_{12}|D) = 0.71$. This is the most complex model considered, with 6 estimated parameters. In Model 12 translocation, NTP binding and catalysis are all kinetic processes; the displacement required to facilitate formation of the translocation transition state, $\delta_1 < \delta$, is estimated ($\hat{\delta_1} = 3.1$ $\angstrom$); and the standard Gibbs energy of the posttranslocated state is influenced by parameter $\Delta G_{\tau 1} \neq 0$. \par

The posterior distribution for  RNAP  consists of the same set models as that of pol II. For RNAP, Model 11 has the highest probability $P(M_{11}|D) = 0.81$. This model is a submodel of Model 12 with one fewer parameter: in Model 11 NTP binding is treated as an equilibrium process while in Model 12 it is not. \par


The only model in the 95 \% credible set for T7 pol is Model 5 $P(M_{5}|D) = 0.96$. In Model 5 (4 parameters) translocation, but not binding, is treated as an equilibrium process, and $\Delta G_{\tau 1}$ is estimated. This positions T7 pol in a quite different area of the model space to the other two polymerases. \par










\subsection*{Translocation rates differ among RNA polymerases} For RNAP and pol II, we estimate that a partial equilibrium approximation for the translocation step is inadequate. The posterior probability that such models are inadequate is 1.00 (see Table \ref{table1}). For T7 pol, however, translocation is significantly faster than catalysis and is best modelled with a partial equilibrium approximation. Using estimates for $\Delta G_\tau^\ddag$ and $\Delta G_{\tau 1}$ under the maximum posterior models (Model 11 for RNAP and Model 12 for pol II) we estimate the mean forward $\bar{k}_{fwd}$ and backward $\bar{k}_{bck}$ translocation rates averaged across the \textit{ rpoB} sequence as:  230 s$^{-1}$ and 112 s$^{-1}$ for RNAP, and 350 s$^{-1}$ and 12.7 s$^{-1}$ for pol II, respectively (3 sf). These estimates are within one order of magnitude of the respective estimate for the rate of catalysis (Fig ~\ref{fig5}) suggesting that translocation and catalysis indeed occur on similar timescales.  \par

For RNAP and pol II, translocation has frequently been modelled as an equilibrium process \cite{abbondanzieri2005direct, bai2004sequence, mejia2015trigger, larson2012trigger, schweikhard2014transcription}, however in some recent analyses this assumption has been rejected \cite{tadigotla2006thermodynamic, maoileidigh2011unified, nedialkov2012rna, Dangkulwanich2013complete, kireeva2018rna}. Our Bayesian analysis supports this. In contrast, there is general agreement that translocation in T7 pol is adequately modelled as an equilibrium process \cite{guajardo1998ntp, arnold2001kinetic, thomen2008t7}. \par





\subsection*{The data does not determine the kinetics of the NTP binding step}





\hl{// Content relocated and adapted from ``Model selection with MCMC-ABC"}


It remains unclear how to best model the NTP binding step. Models that describe NTP binding as a kinetic process have posterior probabilities of $0.19$ for RNAP, $0.71$ for pol II and $0.96$ for T7 pol (Table \ref{table1}). However, in our sensitivity analysis, where we used different a prior distribution for $\frac{k_{rel}}{k_{bind}}$, these probabilities were $0.65$, $0.22$, and $0.19$, respectively (results not shown).


Furthermore, $\frac{k_{rel}}{k_{bind}}$ and $k_{bind}$ (Models 5 and 12) are unable to be estimated simultaneously. For pol II and for T7 pol, $k_{bind}$ is estimated at around 0.48 and 1.4 $\mu M^{-1}$ s$^{-1}$ respectively with fairly narrow 95\% highest posterior density (HPD) intervals (Fig ~\ref{fig5}). However, the HPD interval of $\frac{k_{rel}}{k_{bind}}$ spans three orders of magnitude and the value of this parameter was therefore poorly informed by the data. For RNAP, in contrast, neither $k_{bind}$ nor $\frac{k_{rel}}{k_{bind}}$ were well-informed by the data and both have HPD intervals spanning 1-2 orders of magnitude. This non-identifiability -- where two or more parameters are unable to be estimated simultaneously (\nameref{S4_Appendix}) -- highlights the appeal in an NTP binding equilibrium model where only one parameter $\frac{k_{rel}}{k_{bind}}$ needs to be estimated, despite the unrealistic assumptions it may invoke. In the case of each enzyme, the data has taught us nothing about one or two of the binding parameters.

%unable to be estimated reliably from the data


The pause-free mean velocities measured during transcription elongation follow Michaelis-Menten kinetics
even though the reaction cycle is more complicated than that of a simple enzyme \cite{wong2018structural}. As such, the inability to resolve the timescale of the substrate binding step is unsurprising \cite{briggs1925note, english2006ever, schnell2014validity}.




%Furthermore, our kinetic-binding-model estimates for $\frac{k_{rel}}{k_{bind}}$ and sometimes $k_{bind}$ have HPD intervals which span multiple orders of magnitude (Fig ~\ref{fig5}). Finally, the ESS of $\frac{k_{rel}}{k_{bind}}$, for models which use the parameter, is significantly smaller than the remaining parameters, suggesting the MCMC-ABC encountered mixing problems. Obtaining reliable estimates for this parameter was not practical under limited computational resources.


Overall, these three factors, i) the sensitivity of the posterior probabilities to the choice of prior, ii) the intermediate magnitude of said probabilities, iii) the inability to estimate both $\frac{k_{rel}}{k_{bind}}$ and $k_{bind}$ simultaneously, collectively imply the data carries very little information about the rates of NTP binding and release.




In the transcription literature, NTP binding is almost always assumed to achieve equilibrium for RNAP, pol II, and T7 pol \cite{maoileidigh2011unified, tadigotla2006thermodynamic, bai2004sequence, abbondanzieri2005direct, larson2012trigger, schweikhard2014transcription, Dangkulwanich2013complete, arnold2001kinetic, thomen2008t7, thomen2005unravelling}. However Mejia et al. 2015 \cite{mejia2015trigger} have shown that NTP binding is indeed rate-limiting, and that mutations in the RNAP trigger loop impair the binding rate thus suggesting that the trigger loop is coupled with NTP binding. \par   





\subsection*{RNAP has an energetic preference for the posttranslocated state}
In previous stochastic sequence-dependent models \cite{bai2004sequence, tadigotla2006thermodynamic} the standard Gibbs energies of the pre and posttranslocated states have been based solely on the nucleic acid  basepairing energies. Our models include an additional term, $\Delta G_{\tau 1}$, to account for potential interactions between the protein and the nucleic acid. The marginal posterior probability of a model in which an additional term $\Delta G_{\tau 1}$ is required is 1.00 in all three polymerases. In each case $\Delta G_{\tau 1}$ was estimated to be less than 0 $k_BT$ and 0 $k_BT$ is not included in the 95 \% HPD interval (Fig ~\ref{fig5}). We find that $\Delta \hat{G_{\tau 1}}$ is the most significant in pol II and T7 pol: $-4.6$ $k_BT$  and $-4.0$ $k_BT$ respectively,  while $\Delta \hat{G_{\tau 1}} = -2.0$ $k_BT$ for RNAP (2 sf). \par

These results suggest that structural elements within RNA polymerases can energetically favour posttranslocated states over pretranslocated states. We note that the sequence-dependent contribution of the dangling end of the DNA/RNA hybrid is included in the thermodynamic model. The energetic bias for the posttranslocated state is separable from this effect.  \par

To facilitate comparison with previous deterministic models, using our estimates of $\Delta G_{\tau 1}$ we calculated the equilibrium constant between the pre and posttranslocated states. Geometrically averaged across the \textit{rpoB} gene, these are




\begin{eqnarray}
%\label{eq:deterministicBoltzmann}
\bar{K}_\tau = \frac{1}{L - l_0} exp \{ \sum\limits_{l=l_0}^{L-1} \text{ln} (k_{bck}(l) / k_{fwd}(l)) \} = \begin{cases} 0.77 \text{ for RNAP} \\ 0.057 \text{ for pol II} \\ 0.10 \text{ for T7 pol.}  \end{cases}
\end{eqnarray}


Thus, for all three polymerases, $K_\tau < 1$, indicating that the small energetic preference that the protein has for the posttranslocated state is sufficient to override the loss of basepairing energy, thereby biasing the system towards population of the posttranslocated positions. This is in agreement with estimates made for pol II and T7 pol \cite{larson2012trigger, schweikhard2014transcription, thomen2008t7,yin2004structural, yu2012small} and Kireeva et al. 2018 \cite{kireeva2018rna} for RNAP: \textit{``forward translocation occurs in milliseconds and is poorly reversible''.} However these estimates are inconsistent with some RNAP and pol II studies which place this ratio above 1  \cite{abbondanzieri2005direct, kireeva2008transient, maoileidigh2011unified, Dangkulwanich2013complete}.

Kinetic modelling can itself suggest no physical mechanism for the stabilisation. Yu et al. 2012 \cite{yu2012small} have identified a conserved tyrosine residue near the active site of T7 pol that pushes against the $3^\prime$ end of the mRNA, and thus stabilises the posttranslocated state. They propose a similar mechanism for the multi-subunit RNA polymerases.  \par




\subsection*{$\delta_1$ may be an important parameter but its physical meaning is unclear}

Our results suggest that $\delta_1$, the distance that RNA polymerase must translocate forward by to reach the translocation transition state, is a necessary parameter to estimate for RNAP and pol II. Setting $\delta_1 = \delta / 2$ is not sufficient. The marginal posterior probability of models which estimate this term is 1.00. $\delta_1$ is irrelevant to the modeling of the T7 pol data because the best models invoke a partial equilibrium approximation for the translocation step.




While our prior distribution restricted $\delta_1$ to lie in the range $(0, \delta)$, the upper end our 95\% HPD intervals of $\delta_1$ for RNAP and pol II are very close to $\delta = 3.4$ $\angstrom$. If it was not for this prior distribution, $\delta_1$ estimates would have included values higher than $\delta$. Similar results have been observed by Maoil{\'e}idigh et al. 2011 \cite{maoileidigh2011unified} for RNAP. \par

Our interpretation of $\delta_1$ implies it should never be greater than $\delta$ nor should $\delta$ be more than the width of one basepair. The physical meaning of $\delta_1$ with values greater than $\delta$ is thus unclear. It is noted that $\delta_1$ is only used when $F \neq 0$. \\





\subsection*{Comparing the kinetics of RNA polymerases}




The \textit{ in vivo} rate of transcription elongation varies considerably across  RNAP, pol II and T7 pol. The prokaryotic and eukaryotic RNA polymerases have a mean rate ranging from 20-120 bp/s \cite{iost1992bacteriophage, vogel1994rna, ryals1982temperature, tennyson1995human, darzacq2007vivo, mason2005distinction, kainov2004packaging}, which may be slowed down in histone-wrapped regions of eukaryotic genomes \cite{hodges2009nucleosomal}. In contrast, Bacteriophage T7 pol operates up to an order of magnitude faster (around 200-240 bp/s \cite{iost1992bacteriophage, makarova1995transcribing}) and is known to be quite insensitive to transcriptional pause sites \cite{skinner2004promoter, thomen2008t7}.



In additional to these differences, we have shown that translocation is very rapid in T7 pol, relative to the rate of NTP incorporation, while the disparity is much less significant in RNAP and pol II. Furthermore, the model does not fit the data for T7 pol as closely it does for RNAP and pol II (Fig \ref{fig6}).  T7 pol therefore seems to operate under quite a different kinetic scheme than that of the cellular polymerases, which is not unexpected given their distant evolutionary relationship \cite{sousa1993crystal}. \par




In general, the elongation velocity of RNA polymerase is significantly slower in an optical trap (with estimates ranging from 9.7-22 bp/s for RNAP \cite{neuman2003ubiquitous, davenport2000single, tolic2004diversity, mejia2015trigger, mejia2008thermal})  compared with that of the untethered enzyme (with estimates \textit{ in vitro} or \textit{ in vivo} ranging from 25-118 bp/s for RNAP \cite{iost1992bacteriophage, vogel1994rna, burns1998combinatorial, kingston1981direct}). This relationship holds for multiple RNA polymerases including \textit{E. coli} RNAP, \textit{S. cerevisiae} pol II \cite{galburt2009single, larson2012trigger, Dangkulwanich2013complete, kireeva2008transient}, Bacteriophage T7 pol \cite{skinner2004promoter, thomen2008t7, iost1992bacteriophage, anand2006transient}, and Bacteriophage $\Phi 6$ P2 \cite{dulin2015elongation, usala1980displacement}. This suggests that optical trapping perturbs the system to a significant extent. Additionally, varying degrees of heterogeneity in elongation rate have been observed across different polymerase complexes even under the same conditions \cite{tolic2004diversity, neuman2003ubiquitous, thomen2008t7}. \par

The velocity perturbations resulting from the optical trapping apparatus will be propagated into the model parameters, especially $k_{cat}$, and $\Delta G^\ddag_\tau$, and some caution is needed when extrapolating these results to untethered systems.   





\subsection*{Bayesian inference of transcription elongation} To our knowledge we are the first to perform Bayesian inference on single-molecule models of transcription elongation. This was achieved by simulation which necessitated the use of approximate Bayesian computation. An alternative would be to build and use a likelihood function (ie. the probability of taking exactly $t$ units of time for RNA polymerase to copy the sequence $n$ times). \hl{The latter approach can be achieved using chemical master equations, as opposed to (Gillespie) sampling from the distribution. Finding analytical, stable numerical, or approximate solutions to the chemical master equations could provide a similar insight in less computational time, however is susceptible to a multitude of analytical and numerical issues associated with the exponentiation of an arbitrary transition rate matrix that grows with the length of the sequence} (\nameref{S2_Appendix}) \cite{moler2003nineteen}. \hl{This problem would be amplified by the introduction of backtracking, hypertranslocation, or NTP misincorporation reactions into the model, for instance. The Bayesian framework we have presented, although computationally intensive due to its simulation requirement, is general and will work on any model of transcription without the need to resolve these issues. The path has been paved for modelling transcriptional pausing, for instance} \cite{bai2004sequence,tadigotla2006thermodynamic,bai2010comparison}. \hl{Nevertheless, likelihood-based Bayesian inference is an approach that should be explored in the future. }

%\hl{The chemical master equations for single-cycles of these models are presented in} \nameref{S2_Appendix}.

% which would involve constructing a rate matrix for the transcription elongation pathway and exponentiating the rate matrix to calculate the likelihood (ie. the probability of taking exactly $t$ units of time for RNA polymerase to copy the sequence $n$ times). This would be computationally faster however a multitude of numerical issues can arise from matrix exponentiation when the rate matrix is large, or has an absorbing state. This is an approach which we would like to explore in the future. \par

We have demonstrated that single-molecule data can be usefully analysed using a Bayesian inference and model selection framework. This analysis would have even greater statistical power if applied to the progression of individual RNA polymerase complexes instead of mean velocities averaged across multiple experiments.  \\



\section*{\hl{Conclusion}}


In this article we evaluated some simple Brownian ratchet models of transcription elongation (Fig \ref{fig2}). By varying the parameterisation of the translocation step  (Fig \ref{fig3}) and incorporating partial equilibrium approximations commonly invoked in the literature   (Fig \ref{fig4}A) we enumerated a total of 12 related models (Fig \ref{fig4}B). Using stochastic simulations and approximate Bayesian computation, we then assessed which of these models were capable of describing the force-velocity data previously measured for several RNA polymerases (Table ~\ref{table1} and Fig ~\ref{fig5}) using single-molecule optical trapping experiments \cite{abbondanzieri2005direct, schweikhard2014transcription, thomen2008t7}.

Our analysis suggests that 1) different partial equilibrium approximations of the translocation step are appropriate for the multisubunit RNA polymerases versus the single subunit  T7 RNA polymerase. 2) Treatment of the NTP binding step remains a point of ambiguity. The existing data does not place strong constraints on the modelling of this step. 3) There is an energetic bias for posttranslocated state. 4) The model of the force-dependent translocation, which invokes transition state theory, is not physically realistic.







\section*{Supporting information}



\paragraph*{S1 Appendix.}
\label{S1_Appendix}
\textbf{ Stochastic simulation.}


Reactions are simulated using the Gillespie algorithm \cite{gillespie1977exact}.  Given the current state $s$ and a set of possible reactions $s\rightarrow s_1, s\rightarrow s_2, \dotso, s\rightarrow s_n$ with rate constants $k_1, k_2, \dotso, k_n$, the next reaction to perform is sampled proportional to its rate:

\begin{eqnarray}
    p(s\rightarrow s_i) = \frac{k_i}{\sum\limits_{j=1}^n k_j}.
\end{eqnarray}

The amount of time the reaction takes to occur is sampled from the exponential distribution with rate $\sum\limits_{j=1}^n k_j$. \\








\paragraph*{S2 Appendix.}
\label{S2_Appendix}
\textbf{ Chemical master equations.}

\hl{// Added chemical master equations}

The chemical master equations for the four equilibrium variants (Fig ~\ref{fig4}A) of single nucleotide addition cycles are provided in this section. Transcription of a full gene involves chaining multiple of these single-cycle models together.



\textbf{Translocation and binding equilibrium model} \par


The state pathway of a single-cycle of the translocation and binding equilibrium model is


\begin{eqnarray}
    S(l,0) \xrightleftharpoons[K_\tau(l)]{} S(l,1) \xrightleftharpoons[K_D]{} S_N(l,1) \xrightarrow[k_{cat}]{} S(l+1,0)
\end{eqnarray}


where $S_N$ denotes a state where NTP is bound. As states $S(l,0)$, $S(l,1)$, and $S_N(l,1)$ are in mutual equilibrium they can be coalesced into one state. Let $S(l)$ be the coalesced state that exists in equilibrium between the three, then the pathway may be rewritten as


\begin{eqnarray}
    S(l) \xrightarrow[\omega_{cat}(l)]{} S(l+1,0)
\end{eqnarray}

where $\omega_{cat}(l)$ is the \textit{effective} rate of catalysis from this state. This term is equal to $k_{cat}$ multiplied by the proportion of time that the coalesced state has NTP bound, ie. in state $S_N(l,1)$.


\begin{align}
    \omega_{cat}(l) &= p( S_N(l,1) ) \; k_{cat} \\
    &= \frac{ \frac{[NTP]}{K_D} }{ 1 + \frac{[NTP]}{K_D} + exp\{ -(\Delta G^{(bp)}_{S(l,0)} -  \Delta G^{(bp)}_{S(l,1)} - \Delta G _{\tau 1} ) \} } \; k_{cat}
\end{align}


Let $p(S, \mathbb{T})$ be the probability of the system existing in state $S$ at time $\mathbb{T}$. Under this model, the chemical master equation of a single-cycle is:


\begin{eqnarray}
    \begin{pmatrix}
        \frac{d p( S(l), \mathbb{T} )}{d\mathbb{T}} \\ \\
        \frac{d p( S(l+1,0), \mathbb{T} )}{d\mathbb{T}}
    \end{pmatrix} =
    \begin{pmatrix}
        -\omega_{cat}(l) & 0 \\ \\
        \omega_{cat}(l) & 0
    \end{pmatrix}
    \begin{pmatrix}
        p( S(l), \mathbb{T} ) \\ \\
        p( S(l+1,0), \mathbb{T} )
    \end{pmatrix}
\end{eqnarray}



This system is simple enough to solve analytically. Let $p(\mathbb{T})$ be the Markov transition matrix after time $\mathbb{T}$ and let $Q$ be the Markov process transition rate matrix. In these matrices, entry $i,j$ is the probability ($p$) or rate ($Q$) of transition from $i$ to $j$.


\begin{align}
    p(\mathbb{T}) =& exp\{ Q\mathbb{T} \} \\
    =& exp \Bigg\{
     \begin{pmatrix}
        -\omega_{cat}(l) & \omega_{cat}(l) \\ \\
        0 & 0
    \end{pmatrix} \mathbb{T}
    \Bigg\} \\
    =& \begin{pmatrix}
        e^{-\omega_{cat}(l) \mathbb{T}} & 1 - e^{-\omega_{cat}(l) \mathbb{T}} \\ \\
        0 & 1
    \end{pmatrix}
\end{align}


It is noted that as the time $\mathbb{T}$ approaches infinity, the probability of the system existing in state $S(l+1,0)$ approaches 1, because it is an absorbing state.

Let $f(\mathbb{T})$ be the probability density of taking \textit{exactly} $\mathbb{T}$ units of time to arrive at state $S(l+1,0)$, starting from $S(l)$. Because $S(l+1,0)$ is an absorbing state, computing $f(\mathbb{T})$ is trivial.


\begin{align}
    f(\mathbb{T}) =& \frac{d}{d\mathbb{T}} \; (1 - e^{-\omega_{cat}(l) \mathbb{T}}) \\
     =& \omega_{cat}(l) \; e^{- \omega_{cat}(l) \mathbb{T}}
\end{align}





Transcribing the full gene requires the traversal of all $L - l_0 + 1$ states. The size of the transition rate matrix for transcribing the entire sequence therefore grows with the length of the DNA sequence.


\begin{align}
    Q = \begin{pmatrix}
        -\omega_{cat}(l_0) & \omega_{cat}(l_0) & 0 & 0 & \dotso & 0 \\ \\
        0 & -\omega_{cat}(l_0 + 1) & \omega_{cat}(l_0 + 1) & 0 & \dotso & 0 \\ \\
        \vdots & \ddots & \ddots &  \ddots & \ddots & \vdots\\ \\
        0 & 0 & 0 & 0 & -\omega_{cat}(L - 1) & \omega_{cat}(L - 1) \\ \\
        0 & 0 & 0 & 0 & 0  & 0\\
    \end{pmatrix}
\end{align}


Applying the matrix exponential function to a linear pathway, such as the one described above, has an analytical solution \cite{jahnke2007solving}. However, for non-linear systems with an arbitrary number of states (eg. the other 3 equilibrium model variants), analytical solutions may not exist and numerical solutions likely exhibit numerical instabilities \cite{moler2003nineteen}. This option was not further investigated in this project (simulation was used instead). However, analytical or stable numerical solutions for $f(\mathbb{T})$ for an arbitrary model would facilitate the use of likelihood functions in Bayesian inference, thereby rendering simulation obsolete.



\textbf{Binding equilibrium model} \par


The state pathway of a single-cycle of the binding equilibrium model is


\begin{eqnarray}
S(l,0) \mathop{\rightleftarrows}_{k_{bck(l)}}^{k_{fwd(l)}} S(l,1) \xrightleftharpoons[K_D]{} S_N(l,1) \xrightarrow[k_{cat}]{} S(l+1,0)
\end{eqnarray}


The binding equilibrium assumption permits the coalescence of $S(l,1)$ and $S_N(l,1)$ into a single state $S_N(l)$. Thus, the pathway can be rewritten as


\begin{eqnarray}
S(l,0) \mathop{\rightleftarrows}_{\omega_{bck(l)}}^{k_{fwd(l)}} S_N(l)  \xrightarrow[\omega_{cat}]{} S(l+1,0)
\end{eqnarray}

where $\omega_{bck(l)}$ is the effective rate of backwards translocation from $S_N(l)$, and $\omega_{cat}$ is the effective rate of catalysis. These rates are derived by multiplying their composite rates  -- $k_{bck(l)}$ and $k_{cat}$ -- by the probability of the system existing in the required state to apply the reaction -- $p(S(l,1))$ and $p( S_N(l,1))$.


\begin{align}
    \omega_{bck}(l) &= p( S(l,1) ) \; k_{bck}(l) \\
    &= \frac{1}{ 1 + \frac{[NTP]}{K_D}} \; k_{bck}(l)
\end{align}



\begin{align}
    \omega_{cat}(l) &= p( S_N(l,1) ) \; k_{cat} \\
    &= \frac{ \frac{[NTP]}{K_D} }{ 1 + \frac{[NTP]}{K_D}} \; k_{cat}
\end{align}


Let $p(S, \mathbb{T})$ be the probability of the system being at state $S$ at time $\mathbb{T}$. Under this model, the chemical master equation of a single-cycle is:


\begin{eqnarray}
    \begin{pmatrix}
        \frac{d p( S(l,0), \mathbb{T} )}{d\mathbb{T}} \\ \\
        \frac{d p( S_N(l), \mathbb{T} )}{d\mathbb{T}} \\ \\
        \frac{d p( S(l+1,0), \mathbb{T} )}{d\mathbb{T}}
    \end{pmatrix} =
    \begin{pmatrix}
        -k_{fwd(l)} & \omega_{bck}(l) & 0 \\ \\
        k_{fwd(l)} & -\omega_{bck}(l) - \omega_{cat}(l) & 0 \\ \\
        0 & \omega_{cat}(l) & 0
    \end{pmatrix}
    \begin{pmatrix}
        p( S(l,0), \mathbb{T} ) \\ \\
        p( S_N(l), \mathbb{T} ) \\ \\
        p( S(l+1,0), \mathbb{T} )
    \end{pmatrix}
\end{eqnarray}



\textbf{Translocation equilibrium model} \par


The state pathway of a single-cycle of the translocation equilibrium model is


\begin{eqnarray}
S(l,0) \xrightleftharpoons[K_\tau(l)]{}  S(l,1) \mathop{\rightleftarrows}_{k_{rel}}^{k_{bind}[NTP]} S_N(l,1)  \xrightarrow[k_{cat}]{} S(l+1,0)
\end{eqnarray}



The translocation equilibrium assumption permits the coalescence of $S(l,0)$ and $S(l,1)$ into a single state $S_\tau(l)$. Thus, the pathway can be rewritten as


\begin{eqnarray}
S_\tau(l) \mathop{\rightleftarrows}_{k_{rel}}^{\omega_{bind}[NTP]} S_N(l,1)  \xrightarrow[k_{cat}]{} S(l+1,0)
\end{eqnarray}


where $\omega_{bind}$ is the effective rate of NTP binding.


\begin{align}
    \omega_{bind} &= p( S(l,1) ) \; k_{bind} \\
    &= \frac{ exp \{ -(\Delta G^{(bp)}_{S(l,1)} + \Delta G _{\tau 1}) \} }{ exp \{ -(\Delta G^{(bp)}_{S(l,1)} + \Delta G _{\tau 1}) \} + exp \{ -\Delta G^{(bp)}_{S(l,0)} \}} \; k_{bind}
\end{align}



Let $p(S, \mathbb{T})$ be the probability of the system being at state $S$ at time $\mathbb{T}$. Under this model, the chemical master equation of a single-cycle is:



\begin{eqnarray}
    \begin{pmatrix}
        \frac{d p( S_\tau(l), \mathbb{T} )}{d\mathbb{T}} \\ \\
        \frac{d p( S_N(l,1), \mathbb{T} )}{d\mathbb{T}} \\ \\
        \frac{d p( S(l+1,0), \mathbb{T} )}{d\mathbb{T}}
    \end{pmatrix} =
    \begin{pmatrix}
        -\omega_{bind}[NTP] & k_{rel} & 0 \\ \\
        \omega_{bind}[NTP] & -k_{rel} - k_{cat} & 0 \\ \\
        0 & k_{cat} & 0
    \end{pmatrix}
    \begin{pmatrix}
        p( S_\tau(l), \mathbb{T} ) \\ \\
        p( S_N(l,1), \mathbb{T} ) \\ \\
        p( S(l+1,0), \mathbb{T} )
    \end{pmatrix}
\end{eqnarray}


\textbf{Full kinetic model} \par


The state pathway of a single-cycle of the full kinetic model is


\begin{eqnarray}
S(l,0) \mathop{\rightleftarrows}_{k_{bck}(l)}^{k_{fwd}(l)}  S(l,1) \mathop{\rightleftarrows}_{k_{rel}}^{k_{bind}[NTP]} S_N(l,1)  \xrightarrow[k_{cat}]{} S(l+1,0)
\end{eqnarray}


Let $p(S, \mathbb{T})$ be the probability of the system being at state $S$ at time $\mathbb{T}$. Under this model, the chemical master equation of a single-cycle is:


\begin{eqnarray}
    \begin{pmatrix}
        \frac{d p( S(l,0), \mathbb{T} )}{d\mathbb{T}} \\ \\
        \frac{d p( S(l,1), \mathbb{T} )}{d\mathbb{T}} \\ \\
        \frac{d p( S_N(l,1), \mathbb{T} )}{d\mathbb{T}} \\ \\
        \frac{d p( S(l+1,0), \mathbb{T} )}{d\mathbb{T}}
    \end{pmatrix} =
    \begin{pmatrix}
        -k_{fwd}(l) & k_{bck}(l) & 0 & 0 \\ \\
        k_{fwd}(l) & -k_{bck}(l) - k_{bind}[NTP] & k_{rel} & 0 \\ \\
        0 & k_{bind}[NTP] & -k_{rel} -k_{cat} & 0 \\ \\
        0 & 0 &  k_{cat} & 0
    \end{pmatrix}
    \begin{pmatrix}
        p( S(l,0), \mathbb{T} ) \\ \\
        p( S(l,1), \mathbb{T} ) \\ \\
        p( S_N(l,1), \mathbb{T} ) \\ \\
        p( S(l+1,0), \mathbb{T} )
    \end{pmatrix}
\end{eqnarray}








\paragraph*{S3 Appendix.}
\label{S3_Appendix}



\textbf{ MCMC-ABC.} Given model/parameters $\Theta$ and observed data $D = (D_1, D_2, \dotso, D_n)$, Bayesian inference conventionally involves approximating the posterior probability distribution $P(\Theta | D)$ using the likelihood $P(D|\Theta)$ and the prior $P(\Theta)$.

\begin{eqnarray}
    P(\Theta | D) \propto P(D | \Theta) P(\Theta).
\end{eqnarray}

As there is no easily computed likelihood function, simulation is used. The chi-squared test statistic $X^2$  evaluates how well a given set of parameters fits the data.

\begin{eqnarray}
    X^2 = \sum\limits_{i} \frac{(S_i - D_i)^2}{S_i}
\end{eqnarray}

where $S_i$ is the mean velocity simulated under the same [NTP] and applied force $F$ that $D_i$ was measured under. The probability that $X^2=0$ is equal to the likelihood $P(D | \Theta)$, however it is computationally impractical to only accept parameters into the posterior distribution when the simulation yields $X^2=0$. Therefore a threshold $\epsilon$ is used and sample $\Theta_i$ is accepted into the posterior only if $X^2(\Theta_i) \leq \epsilon$. This method is called approximate Bayesian computation \cite{beaumont2010approximate, csillery2010approximate}. This is coupled with Markov chain Monte Carlo (MCMC) to give the MCMC-ABC algorithm which is becoming increasingly popular among computational biologists \cite{beaumont2010approximate, ratmann2007using}.  \\


\hl{// Content below relocated and adapted from the since-removed ``Materials and Methods"}




Each MCMC chain estimated six parameters and the model indicator $M$. This means the 12 models share the same parameter objects in the MCMC. There are therefore seven terms to estimate: $M$, $k_{cat}$, $\frac{k_{rel}}{k_{bind}}$, $k_{bind}$, $\Delta G_{\tau 1}$, $\Delta G^\ddag_\tau$, and $\delta_1$.


When the current model $M_i$ does not use a parameter (eg. in Model 5 $\delta_1$ is not used), the parameter is still estimated even though it is not being used. When the model requires a parameter to be held constant (eg. in Model 1 $\Delta G_{\tau 1} = 0$), the parameter is set to its constant during the simulation. This is done without affecting or being affected by its current estimate, which is used by other models.


%For each RNA polymerase we performed one MCMC which, alongside the parameters, additionally estimated model indicator $M$. Each elongation model uses up to 6 parameters in its simulation component. This means that the 12 models are sharing the same set of estimates in the MCMC. There are therefore seven terms for this MCMC to estimate: $M$, $k_{cat}$, $\frac{k_{rel}}{k_{bind}}$, $k_{bind}$, $\Delta G_{\tau 1}$, $\Delta G^\ddag_\tau$, and $\delta_1$. When current model $M_i$ requires a parameter to be held constant (eg. in Model 1 $\Delta G_{\tau 1} = 0$), then this parameter would be set to its constant during the simulation without affecting or being affected by its current MCMC estimate.




To achieve convergence, we used an exponential cooling scheme on $\epsilon$ \cite{ratmann2007using} where $\epsilon_{i+1} = \max(\epsilon_{min}, \epsilon_{i}\gamma)$ for manually tuned values of $0 < \gamma < 1$ and $\epsilon_0$. Chains which failed to converge were discarded. A heavy-tailed distribution \cite{brewer2016dnest4} is used as a proposal function, and the parameter to change at each step in the MCMC is selected uniformly at random.


We ran one or more independent MCMC-ABC chains for each selected $\epsilon_{min}$ / RNA polymerase combination.
Selecting the threshold $\epsilon_{min}$ is a critical process in approximate Bayesian computation. Threshold $\epsilon_{min}$ must be large enough to achieve convergence within finite computational resources, but small enough that the resulting posterior distribution is still an accurate approximation of the true posterior distribution. For each RNA polymerase we set $\epsilon_{min}$ to some initial guess. Then we ran the MCMC chain until the ESS for $X^2$ was large ($>$ 300) and lowered $\epsilon_{min}$ to the bottom 0.05 quantile of the posterior distribution of $X^2$. This step was repeated until either: a) the distribution of model indicators $M$ converged (model posterior probabilities have changed by less than 0.01, on average). Or, b) the acceptance rate was less than 5\%. The values of $\epsilon_{min}$ used in the final posterior distributions were 2.39 for RNAP, 0.705 for pol II, and 4.63 for T7 pol (Table ~\ref{table1}).




Parameter point estimates (Fig ~\ref{fig5}) are the geometric median: that is the value which minimises the total Euclidean distance from the other posterior samples. Parameters were normalised into z-scores first. Our code is open source and available at http://www.polymerase.nz. Textfiles containing the posterior distributions and simulation settings are available to download or visualise with the software.








\paragraph*{S4 Appendix.}
\label{S4_Appendix}
\textbf{ Prior distributions.}

\textbf{Prior for $\Delta G^\ddag_{\tau}$, which governs the rates of translocation} \par



RNAP/pol II: to select a prior for $\Delta G^\ddag_{\tau}$ we simulated transcription on the \textit{ rpoB} gene under Model 3 -- the simplest binding equilibrium model. $\Delta G^\ddag_{\tau}$ and $k_{cat}$ were sampled uniformly from a relevant range, with $K_D$ held constant at 100 $\mu M$ and [NTP] = 1000 $\mu M$. For each simulation, the mean elongation velocity was calculated. The results are displayed in \nameref{S1_Fig}.


This plot shows that as the energy barrier of translocation ($\Delta G^\ddag_{\tau}$) increases, the velocity decreases. If $\Delta G^\ddag_{\tau} \gtrsim 8$ $k_BT$ then it becomes impossible to achieve a realistic mean velocity, providing a relatively clear upper bound on this parameter. If $\Delta G^\ddag_{\tau} \lesssim 3$ $k_BT$ then translocation becomes very rapid and the same distribution of velocities is obtained in simulations, irrespective of the exact value of $\Delta G^\ddag_{\tau}$. In this case catalysis becomes strongly rate-limiting, and it would be appropriate to apply a partial  equilibrium approximation to the translocation step. This provides an effective lower bound for parameter $\Delta G^\ddag_{\tau}$. Therefore we centered our prior distribution for $\Delta G^\ddag_{\tau}$ in this interval (a normal distribution with a mean of 5.5 and a standard deviation of 0.97, so that the central 99\% interval is (3, 8)).  We performed the same analysis with different values of $K_D$, as well as varying $\Delta G_{\tau 1}$, and arrived at the same interval for $\Delta G^\ddag_{\tau}$ (results not shown).


T7 pol: the same analysis was performed, however with $\Delta G^\ddag_{\tau}$ at its prior mean of $-3.3$ $k_BT$ (\nameref{S1_Fig}).



\textbf{Prior for $k_{bind}$, which governs the rate of NTP binding} \par



To select a prior for  $k_{bind}$ we performed similar simulations, but instead used Model 2 -- the simplest kinetic binding model. $k_{bind}$ and $k_{cat}$ were sampled uniformly from relevant ranges, $K_D$ was set to 100 $\mu$M and [NTP] = 1000 $\mu M$. (\nameref{S1_Fig}).


Depending on the exact value of $k_{cat}$, if $k_{bind} \lesssim 0.1$ $\mu$M$^{-1}$ s$^{-1}$, then it is impossible to achieve a realistic velocity, providing a relatively clear lower bound on this parameter. If $k_{bind} \gtrsim 5$ $\mu$M$^{-1}$ s$^{-1}$ then binding becomes very rapid and the same distribution of velocities is obtained in simulations, irrespective of the exact value of $k_{bind}$. Again this is because catalysis becomes strongly rate limiting in this region, and it would be appropriate to apply a partial  equilibrium approximation to the binding step. Hence we centered our (lognormal) prior around the interval (0.01, 5) -- the conservatively selected bounds reflecting that the experimental data has been collected at differing NTP concentrations, altering the rate. Performing the same analysis with different parameters gave us a similar prior.



\textbf{Prior distribution related to rate of NTP release} \par


A model is non-identifiable if two or more parameterisations can produce the same output. Our preliminary results suggested non-identifiability between $\frac{k_{rel}}{k_{bind}}$ and $k_{bind}$ (\nameref{S1_Fig}). When $k_{bind}$ is low (and hence binding is rate-limiting), there is an approximately linear relationship between $\frac{k_{rel}}{k_{bind}}$ and $k_{bind}$. As $k_{bind}$ increases from 0, the dissociation constant $\frac{k_{rel}}{k_{bind}}$ must also increase in order for the system to achieve the same velocity. However, as binding comes closer to achieving equilibrium, $\frac{k_{rel}}{k_{bind}}$ converges. Most previous estimates of $K_D$ have assumed binding to be at equilibrium. This assumption restrains the values which $K_D$ may take, and subsequently estimates for $K_D$ are typically in the order of $10^1$ - $10^2$ $\mu$M. However for a model in which binding is slow it is expected that estimates of $\frac{k_{rel}}{k_{bind}}$ can be lower. This has indeed been demonstrated by Mejia et al. 2015 \cite{mejia2015trigger} who estimated $\frac{k_{rel}}{k_{bind}}$ to be $0.6$ $
\mu$M. Therefore the prior distribution for $\frac{k_{rel}}{k_{bind}}$ must permit both of these binding models to be tested fairly during Bayesian inference. We centered our lognormal prior for $\frac{k_{rel}}{k_{bind}}$ around a very broad range, with a central 95\% interval of (0.2, 200).

It is noted that selecting a prior distribution which does not discriminate between the kinetic and equilibrium binding models \textit{a priori} may not be plausible.


%\begin{changemargin}{-5in}{0in}



%\end{changemargin}




% Include only the SI item label in the paragraph heading. Use the \nameref{label} command to cite SI items in the text.
\paragraph*{S1 Fig.}
\label{S1_Fig}
{\textbf{ Simulations of the elongation pathway.} Each point is a single simulation of the full \textit{rpoB} gene (4029 nt). For (A-C), Parameters on the x- and z-axis are sampled uniformly at random from the displayed range at the beginning of each trial. The y-axis of each plot (mean elongation velocity) is then measured from the respective simulation. [NTP] and $F$ held constant at 1000 $\mu$M and 0 pN respectively. (A) and (B): Relationship between $\Delta G^\ddag_{\tau}$ and $k_{cat}$ for the melting model with binding at equilibrium (Model 8). $\Delta G_{\tau 1}$ set to its prior mean (0 for RNAP and pol II, and -3.3 for T7 pol). (C) Relationship between $k_{bind}$ and $k_{cat}$ for the kinetic binding model with translocation at equilibrium (Model 2). (D) Relationship between $K_D$ and $k_{bind}$ with translocation held at equilibrium (Model 2). $K_D$ and $k_{bind}$ sampled uniformly from specified range and velocity is measured. Samples with simulated velocities outside of the range 1-2 bp/s were discarded. [NTP] = $10$ $\mu$M and $k_{cat} = 100$ s$^{-1}$.






\section*{Acknowledgments}

We wish to acknowledge the contribution of NeSI high-performance computing facilities to the results of this research. NZ's national facilities are provided by the NZ eScience Infrastructure and funded jointly by NeSI's collaborator institutions and through the Ministry of Business, Innovation \& Employment's Research Infrastructure programme. URL https://www.nesi.org.nz.
\nolinenumbers

% Either type in your references using
% \begin{thebibliography}{}
% \bibitem{}
% Text
% \end{thebibliography}
%
% or
%
% Compile your BiBTeX database using our plos2015.bst
% style file and paste the contents of your .bbl file
% here. See http://journals.plos.org/plosone/s/latex for
% step-by-step instructions.
%

\bibliography{references}



\end{document}












